### PR TITLE
Add a script to publish a folder of binaries as a GH release

### PR DESCRIPTION
**Pull request by DHowett**
This is a crazy fever dream.

I thought, "what if I could just have a script make all the releases, tags and names and upload all the assets to the right place?"

So, here's that script. Check it.

image

image

### CODE REVIEW

**Comment by DHowett** [comment]
For fun, it can also be used to simply inspect a folder of releases:

```
$rcs=.\Draft-TerminalReleases.ps1 ./115 -dumpoutput
(dhowett-sl) ~\src\terminal\tools\ReleaseEngineering % $rcs.Release[0]

Name           : Windows Terminal
Branding       : Release
TagVersion     : 1.14.1962.0
DisplayVersion : 1.14.196
Branch         : release-1.14
Assets         : {Microsoft.WindowsTerminal, Microsoft.WindowsTerminal, Microsoft.WindowsTerminal,
                 Microsoft.WindowsTerminal}
```

```
% $rcs.Release[0].Assets | Format-Table

Name                      Version                  Type Branding WindowsVersion Path
----                      -------                  ---- -------- -------------- ----
Microsoft.WindowsTerminal 1.14.1962.0 ApplicationBundle  Release 10.0.19041.0   C:\Users\duhowett\src\terminal\tools\R…
Microsoft.WindowsTerminal 1.14.1962.0     PreinstallKit  Release 10.0.19041.0   C:\Users\duhowett\src\terminal\tools\R…
Microsoft.WindowsTerminal 1.14.1963.0 ApplicationBundle  Release 10.0.22000.0   C:\Users\duhowett\src\terminal\tools\R…
Microsoft.WindowsTerminal 1.14.1963.0     PreinstallKit  Release 10.0.22000.0   C:\Users\duhowett\src\terminal\tools\R…
```

```
% $rcs.Release | ConvertTo-Json -Depth 5
[
  {
    "Name": "Windows Terminal",
    "Branding": 1,
    "TagVersion": {
      "Major": 1,
      "Minor": 14,
      "Build": 1962,
      "Revision": 0,
      "MajorRevision": 0,
      "MinorRevision": 0
    },
    "DisplayVersion": {
      "Major": 1,
      "Minor": 14,
      "Build": 196,
      "Revision": -1,
      "MajorRevision": -1,
      "MinorRevision": -1
    },
```

**Comment by DHowett** [comment]
Consider renaming the preinstall kit to remove `Windows10_` from it.

**Comment by DHowett** [comment]
This is great and amazing, but it needs to be fixed for portable/zip builds and stuff too.

**Review by lhecker — APPROVED** [review]

**Review by zadjii-msft — APPROVED** [review]

File: tools/ReleaseEngineering/Draft-TerminalReleases.ps1 [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+
+	[Asset[]]$Assets
+
+	Release([Asset[]]$a) {
+		$this.Assets = $a
+		$this.Branding = $a[0].Branding
+		$this.Name = Switch($this.Branding) {
+			Release { "Windows Terminal" }
+			Preview { "Windows Terminal Preview" }
+			Default { throw "Unknown Branding for release publication $_" }
```
[the remark this answers is not in the document]
Comment (DHowett) [reply tools/ReleaseEngineering/Draft-TerminalReleases.ps1:181]: Hard error if we try to publish Canary = OK with me. We should never publish one of those!

**Pull request merged**